### ISSUE TITLE
Nginx + OpenSSL 1.1.0b doesn't connect to same servers

### ISSUE THREAD

**Issue by catap** [openssl/openssl#1851]
Good day!

I tried build Nginx with last version of openssl (1.1.0b) and it can't connect by HTTPS to upstream (195.154.47.172) with errors:
```
error:0407008A:rsa routines:RSA_padding_check_PKCS1_type_1:invalid padding
error:04067072:rsa routines:rsa_ossl_public_decrypt:padding check failed
error:1416D07B:SSL routines:tls_process_key_exchange:bad signature
```

If I build nginx with OpenSSL 1.0.2j, everything works fine.

I did research (git bisect) and found that it was broken since commit: https://github.com/openssl/openssl/commit/94777c9c86a2b2ea2726c49d6c8f61078558beba

I [attached](https://github.com/openssl/openssl/files/571748/scripts.zip) a bisect script and nginx.conf that I used to find this commit.

**Comment by mattcaswell** [comment]
This is a possible duplicate of #1799, which was recently fixed in git in commit 0f6c9d7.

Please can you try the latest 1.1.0 git branch and see if the problem persists?

**Comment by catap** [comment]
@mattcaswell confirmed!

Thanks!
